### PR TITLE
✨ public-api: Get and Update jobs metadata

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

Now a user can attach a key-values map (denoted "custom metadata") to an existing job.

- ✨ Public-api implements `Get` and `Update` jobs metadata.  
- ✨ Enhanced OAS: added error responses
- ♻️ cleanup w/ ruff both code and tests
- ♻️ cleanup test fixtures

## Related issue/s

- resolves https://github.com/ITISFoundation/osparc-simcore/issues/4313

## How to test

- Driving test: services/api-server/tests/unit/api_solvers/test_api_routers_solvers_jobs_metadata.py

## DevOps  

None

### CODE REVIEW

**Review by GitHK — APPROVED** [review]

File: packages/postgres-database/src/simcore_postgres_database/utils_aiosqlalchemy.py [hunk 1]
```diff
@@ -27,6 +27,5 @@ async def raise_if_migration_not_ready(engine: AsyncEngine):
         )
         head_version_num = get_current_head()
         if version_num != head_version_num:
-            raise DBMigrationError(
-                f"Migration is incomplete, expected {head_version_num} but got {version_num}"
-            )
+            msg = f"Migration is incomplete, expected {head_version_num} but got {version_num}"
+            raise DBMigrationError(msg)
```
Comment (matusdrobuliak66) [inline packages/postgres-database/src/simcore_postgres_database/utils_aiosqlalchemy.py:31]: This is done by ruff automatically?

File: api/specs/webserver/openapi-projects-metadata.yaml [hunk 2]
```diff
@@ -49,45 +49,45 @@ paths:
 components:
   schemas:
     Envelope_ProjectMetadataGet_:
+      title: Envelope[ProjectMetadataGet]
+      type: object
```
Comment (matusdrobuliak66) [inline api/specs/webserver/openapi-projects-metadata.yaml:53]: The same happened to me with the project comments, I dont know why the ordering changed

**Review by matusdrobuliak66 — APPROVED** [review]

File: services/api-server/src/simcore_service_api_server/api/dependencies/database.py [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import logging
-from typing import AsyncGenerator, Callable
```
Comment (bisgaard-itis) [inline services/api-server/src/simcore_service_api_server/api/dependencies/database.py:2]: Question for my learning: Why this change?

**Review by bisgaard-itis — APPROVED** [review]
Looks good to me

File: packages/postgres-database/src/simcore_postgres_database/utils_aiosqlalchemy.py (hunk 1 above)
Comment (pcrespov) [reply packages/postgres-database/src/simcore_postgres_database/utils_aiosqlalchemy.py:31]: ruff does it

File: services/api-server/src/simcore_service_api_server/api/dependencies/database.py (hunk 3 above)
Comment (pcrespov) [reply services/api-server/src/simcore_service_api_server/api/dependencies/database.py:2]: ruff suggestion. python is moving these classes to collections in later version

File: services/api-server/tests/mocks/cleanup.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
+import json
+import re
+from pathlib import Path
+
+from faker import Faker
+
+_fake = Faker()
```
[the remark this answers is not in the document]
Comment (pcrespov) [reply services/api-server/tests/mocks/cleanup.py:7]: and is `_logger` or `_LOGGER`? :-)

As `constant`, in this context, i understand only non-mutable types like strings or ints. 

https://github.com/ITISFoundation/osparc-simcore/blob/bf838aa1a559f060428536b17ad05e5876129982/docs/coding-conventions.md#python
